lib/src/services/coin_service.dart

```diff
@@ -0,0 +1,14 @@
+import 'package:http/http.dart' as http;
+
+import '../models/coins.dart';
+
+abstract class CoinService {
+  final http.Client client;
```

Comment:
do we need to have this client as a property?

Reply:
We use this client used to override this property to call API in app_coin_services.  By the way, we did the same way based on Mr. Hai's example.